request.go

```diff
@@ -111,6 +111,12 @@ func (s *state) setListerAt(la ListerAt) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	if s.listerAt != nil {
+		if closer, ok := s.listerAt.(CloserListerAt); ok {
+			closer.Close()
```

Comment:
Do we really want to be closing this _now_ at the `setListerAt`?

Reply:
just defensive programming. I think listerAt will be always nil here